allennlp/training/trainer.py

```diff
@@ -56,6 +56,12 @@ def __init__(
         local_rank: int = 0,
         world_size: int = 1,
     ) -> None:
+        if cuda_device == -2:
```

Comment:
I like the feature that this is adding.  I'm not thrilled with using `-2` to specify it; it's too magic.  At the very least this needs better documentation.  I'd be happier if there were a better value to pass in to specify this.  E.g., you could do some string value that at least has a meaningful name, though I'm not super happy with mixing strings and ints here, either.

Reply:
I'd probably be happiest if the default for this setting is `None`, and if you don't specify anything in the config it auto-detects.

Reply:
That seems totally reasonable to me.  I was going to suggest using `None` to mean that, but it was weird with a default value of `-1`.  If you switch the default to `None`, it works out.

Reply:
I made that change. Can you review again?